.circleci/config.yml

```
@@ -20,31 +20,45 @@ e2eTest: &e2eTest
       command: ./e2e-harness teardown
       when: always
 
+orbs:
+  architect: giantswarm/architect@0.6.0
 
-
-version: 2
+version: 2.1
 jobs:
   build:
     machine: true
     steps:
-    - checkout
-    - run:
-        name: Install architect
-        command: |
+      - checkout
+
+      - run: |
           wget -q $(curl -sS -H "Authorization: token $RELEASE_TOKEN" https://api.github.com/repos/giantswarm/architect/releases/latest | grep browser_download_url | head -n 1 | cut -d '"' -f 4)
           chmod +x ./architect
           ./architect version
-    - run:
-        name: Build using architect
-        command: ./architect build
-    - store_test_results:
-        path: /tmp/results
-    - deploy:
-        name: Deploy using architect
-        command: |
-          if [ "${CIRCLE_BRANCH}" == "master" ]; then
-            ./architect deploy
-          fi
+      - run: |
+          date +"%Y" > /tmp/year
+          date +"%m" > /tmp/month
+          date +"%d" > /tmp/day
+      - restore_cache:
+          keys:
+            - go-cache-v1-{{ checksum "/tmp/year" }}-{{ checksum "/tmp/month" }}-{{ checksum "/tmp/day" }}
+            - go-cache-v1-{{ checksum "/tmp/year" }}-{{ checksum "/tmp/month" }}
+            - go-cache-v1-{{ checksum "/tmp/year" }}
+
+      - run: ./architect build
```

We should have step names, see https://github.com/giantswarm/fmt/blob/master/circleci/all-circleci-best-practices.md.
For example:
```suggestion
      - run:
          name: Build using architect
          command: ./architect build
```